starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,610 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(number1, number2) {
+  if (number1 > number2) {
+    return number1;
+  } else if (number2 > number1) {
+    return number2;
+  }
+  return number1;
+}
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(words) {
+  if (words.length === 0) {
+    return null;
+  }
+
+  let longestWord = words[0];
+  for (let i = 0; i < words.length; i++) {
+    if (
+      longestWord.length > words[i].length ||
+      longestWord.length === words[i].length
+    ) {
+      continue;
+    } else {
+      longestWord = words[i];
+    }
+  }
+  return longestWord;
+}
+
+//same function with a for...of loop
+/*function findLongestWord(words) {
+  if (words.length === 0) {
+    return null;
+  }
+
+  let longestWord = words[0];
+  for (word of words) {
+    if (
+      longestWord.length > word.length ||
+      longestWord.length === word.length
+    ) {
+      continue;
+    } else {
+      longestWord = word;
+    }
+  }
+  return longestWord;
+}*/
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(numbers) {
+  let sum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+    sum += numbers[i];
+  }
+  return sum;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numbers) {
+  let sum = 0;
+  let average = null;
+  for (let i = 0; i < numbers.length; i++) {
+    sum += numbers[i];
+  }
+  if (sum !== 0) {
+    average = sum / numbers.length;
+  }
+  return average;
+}
```

Comment:
In this function you are basically repeating the previous function you could have replaces the for loop with a:
```
sum = sumArray(numbers)
```

Besides that all good